components/director/internal/domain/runtime/repository_test.go

```diff
@@ -104,10 +168,21 @@ func TestPgRepository_List_WithFiltersShouldReturnRuntimeModelsForRuntimeEntitie
 		AddRow(runtime1ID, tenantID, "Runtime ABC", "Description for runtime ABC", "INITIAL", timestamp, agentAuthStr).
 		AddRow(runtime2ID, tenantID, "Runtime XYZ", "Description for runtime XYZ", "INITIAL", timestamp, agentAuthStr)
 
-	sqlMock.ExpectQuery(`^SELECT (.+) FROM "public"."runtimes" WHERE "tenant_id" = \$1  AND "id" IN \(SELECT "runtime_id" FROM "public"."labels" WHERE "runtime_id" IS NOT NULL AND "tenant_id" = '` + tenantID + `' AND "key" = 'foo'\)$`).
+	sqlQuery := fmt.Sprintf(`^SELECT (.+) FROM "public"."runtimes" 
+								WHERE "tenant_id" = \$1  AND "id" IN
+									\(SELECT "runtime_id" FROM "public"."labels" WHERE "runtime_id" IS NOT NULL AND "tenant_id" = '`+tenantID+`' AND "key" = 'foo'\)
+									ORDER BY "id" LIMIT %d OFFSET 0$`, rowSize)
+
+	sqlMock.ExpectQuery(sqlQuery).
 		WithArgs(tenantID).
 		WillReturnRows(rows)
 
+	countRows := sqlMock.NewRows([]string{"count"}).AddRow(rowSize)
+
+	sqlMock.ExpectQuery(`^SELECT COUNT \(\*\) FROM "public"."runtimes" WHERE "tenant_id" = \$1$`).
```

Comment:
this is probably an error in the business logic, if you have filters, you should also have them in count query.